src/ewatercycle/container.py

```diff
@@ -0,0 +1,97 @@
+"""Container utilities."""
+from pathlib import Path
+from typing import Dict, Iterable, Literal, Mapping, Optional, Union
+
+from bmipy import Bmi
+from grpc import FutureTimeoutError
+from grpc4bmi.bmi_client_docker import BmiClientDocker
+from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+from grpc4bmi.bmi_memoized import MemoizedBmi
+from grpc4bmi.bmi_optionaldest import OptionalDestBmi
+
+from ewatercycle import CFG
+
+ContainerEngines = Literal["docker", "singularity"]
+"""Supported container engines."""
+
+ImageForContainerEngines = Dict[ContainerEngines, str]
+"""Container image name for each container engine."""
+
+VersionImages = Mapping[str, ImageForContainerEngines]
+"""Dictionary of versions of a model.
+
+Each version has the image name for each container engine.
+"""
+
+
+def start_container(
+    work_dir: Union[str, Path],
+    image_engine: ImageForContainerEngines,
+    input_dirs: Optional[Iterable[str]] = None,
+    image_port=55555,
+    timeout=None,
+    delay=0,
+) -> Bmi:
```

Comment:
Having this isolated from the models is great!
Would it be possible to separate the image selection from the actual starting of container? I'd like to be able to start a container by simply passing in an image (assuming that I know which container engine I'm using)

Reply:
I added start_apptainer_container() and start_docker_container()